src/control/lib/netdetect/netdetect_test.go

```diff
@@ -266,48 +267,33 @@ func TestDeviceAlias(t *testing.T) {
 			topology: "testdata/wolf-133.xml",
 			expected: "hfi1_1",
 		},
-		"i40iw1 alias reverse lookup (boro 84 system topology)": {
-			device:   "i40iw1",
-			topology: "testdata/boro-84.xml",
-			expected: "eth0",
-		},
-		"i40iw0 alias reverse lookup (boro 84 system topology)": {
-			device:   "i40iw0",
-			topology: "testdata/boro-84.xml",
-			expected: "eth1",
-		},
-		"hfi1_0 alias reverse lookup (boro 84 system topology)": {
-			device:   "hfi1_0",
-			topology: "testdata/boro-84.xml",
-			expected: "ib0",
-		},
-		"i40iw1 alias reverse lookup (wolf 133 system topology)": {
-			device:   "i40iw1",
-			topology: "testdata/wolf-133.xml",
-			expected: "eth0",
+		"ib0 alias lookup (multiport-hfi system topology)": {
+			device:   "ib0",
+			topology: "testdata/multiport_hfi_topology.xml",
+			expected: "mlx4_0",
 		},
-		"i40iw0 alias reverse lookup (wolf 133 system topology)": {
-			device:   "i40iw0",
-			topology: "testdata/wolf-133.xml",
-			expected: "eth1",
+		"enp2s0 alias lookup (multiport-hfi system topology)": {
+			device:   "enp2s0",
+			topology: "testdata/multiport_hfi_topology.xml",
+			expected: "mlx4_0",
 		},
-		"hfi1_0 alias reverse lookup (wolf 133 system topology)": {
-			device:   "hfi1_0",
-			topology: "testdata/wolf-133.xml",
-			expected: "ib0",
+		"ib1 alias lookup (multiport-hfi system topology": {
+			device:   "ib1",
+			topology: "testdata/multiport_hfi_topology.xml",
+			expected: "mlx4_1",
 		},
-		"hfi1_1 alias reverse lookup (wolf 133 system topology)": {
-			device:   "hfi1_1",
-			topology: "testdata/wolf-133.xml",
-			expected: "ib1",
+		"ib2 alias lookup (multiport-hfi system topology)": {
+			device:   "ib2",
+			topology: "testdata/multiport_hfi_topology.xml",
+			expected: "mlx4_1",
 		},
 	} {
 		t.Run(name, func(t *testing.T) {
 			_, err := os.Stat(tc.topology)
 			AssertEqual(t, err, nil, "unable to load xmlTopology")
 			os.Setenv("HWLOC_XMLFILE", tc.topology)
 			defer os.Unsetenv("HWLOC_XMLFILE")
-			deviceAlias, err := GetDeviceAlias(tc.device)
+			deviceAlias, err := mockGetDeviceAlias(tc.device, mockSystemDevices)
```

Comment:
I am a bit confused about this change. We seem to now be testing a mock function instead of testing the production code.

Reply:
The GetDeviceAlias function is an all in one function that is self contained.  It minimizes the work required by the caller by requiring only a device name be passed in.  Internally, it determines the system devices by name, a component necessary for proper operation.  However, that system device list depends on actual hardware.  The mockGetDeviceAlias function runs identically to GetDeviceAlias with the difference that it is given a system device list to use.  In that way, there is no dependence on the underlying hardware.  It would be possible to modify GetDeviceAlias to use a system device list passed into it (i.e. like mockGetDeviceAlias) but it would require yet another layer to create that system dependent system device list and supply it.  I considered modifying the function to do this but wasn't convinced that was worth it so I left it as you see it.

Reply:
IMO it would be much better to refactor out a helper function that accepts the system list passed in -- you could probably just rename mockGetDeviceAlias to something like getDeviceAliasWithSystemList and use it directly. Then, the signature for GetDeviceAlias could remain the same, but it would be a thin layer. It would get the real hardware list and then call the helper, which could still be directly tested in this test. Then there's no need to maintain a separate "test" version of production logic.

Reply:
I just submitted a version that I thought was good, but, I decided I like your approach better.  I am updating it now and will resubmit.  You can hold off on any review until I resubmit.